moveit_ros/planning/planning_pipeline/include/moveit/planning_pipeline/planning_pipeline_middleware_handle.hpp

```diff
@@ -0,0 +1,132 @@
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Anthony Baker */
+
+#pragma once
+#include <moveit/planning_pipeline/planning_pipeline.h>
+#include <rclcpp/rclcpp.hpp>
+#include <string>
+
+namespace planning_pipeline
+{
+class PlanningPipelineMiddlewareHandle : public PlanningPipeline::MiddlewareHandle
+{
+public:
+  PlanningPipelineMiddlewareHandle(const rclcpp::Node::SharedPtr& node);
+
+  /**
+   * @brief     Create a display path publisher
+   * @param[in] topic the topic
+   */
+  void createDisplayPathPublisher(const std::string& topic, bool flag) override;
+
+  /**
+   * @brief     Create a received request publisher
+   * @param[in] topic the topic
+   */
+  void createReceivedRequestPublisher(const std::string& topic, bool flag) override;
+
+  /**
+   * @brief     Create a contacts publisher
+   * @param[in] topic the topic
+   */
+  void createContactsPublisher(const std::string& topic, bool flag) override;
+
+  std::string getContactsTopicName() const override;
+
+  /**
+   * @brief     Reset a display path publisher
+   */
+  void resetDisplayPathPublisher() override;
+
+  /**
+   * @brief     Reset a received request publisher
+   */
+  void resetReceivedRequestPublisher() override;
+
+  /**
+   * @brief     Reset a contacts publisher
+   */
+  void resetContactsPublisher() override;
+
+  bool has_parameter(const std::string& name) const override;
+
+  void get_parameter(const std::string& pluginName, std::string& parameter) const override;
+
+  /**
+   * @brief     Create a planner plugin
+   * @param[in] name plugin name
+   */
+  void createPlannerPlugin(const std::string& name, const moveit::core::RobotModelConstPtr& robot_model) override;
```

Comment:
This is an opportunity to improve parameter naming. Since `name` is actually the name of the plugin, it would make sense to call it `plugin_name` instead.